city_scrapers/spiders/il_finance_authority.py

```diff
@@ -0,0 +1,171 @@
+
+            yield scrapy.Request(
+                response.urljoin(pdf_link),
+                callback=self._parse_schedule,
+                dont_filter=True,
+                meta={"title": title, "date": date},
+            )
+
+    def _parse_schedule(self, response):
+        """Parse PDF and then yield to meeting items"""
+        pdf_text = self._parse_agenda_pdf(response)
+        location = self._parse_location(pdf_text)
+        time = self._parse_start(pdf_text)
+        meeting_dict = dict()
+        meeting_dict["title"] = response.meta["title"]
+        meeting_dict["date"] = response.meta["date"]
+        meeting_dict["location"] = location
+        meeting_dict["time"] = time
+
+        yield scrapy.Request(
+            response.url,
+            callback=self._parse_meeting,
+            dont_filter=True,
+            meta={"meeting_dict": meeting_dict},
+        )
+
+    def _parse_agenda_pdf(self, response):
+        try:
+            lp = LAParams(line_margin=0.1)
+            out_str = StringIO()
+            extract_text_to_fp(
+                inf=BytesIO(response.body),
+                outfp=out_str,
+                maxpages=1,
+                laparams=lp,
+                codec="utf-8",
+            )
+
+            pdf_content = out_str.getvalue().replace("\n", "")
+            # Remove duplicate spaces
+            clean_text = re.sub(r"\s+", " ", pdf_content)
+            # Remove underscores
+            clean_text = re.sub(r"_*", "", clean_text)
+            return clean_text
+
+        except PDFSyntaxError as e:
+            print("~~Error: " + str(e))
+
+    def _parse_meeting(self, response):
+        meeting_dict = response.meta["meeting_dict"]
+        title = meeting_dict["title"]
+        date = meeting_dict["date"]
+        time = meeting_dict["time"]
+        location = meeting_dict["location"]
+
+        meeting = Meeting(
+            title=title,
+            description="",
+            classification=self._parse_classification(title),
+            start=self._meeting_datetime(date, time),
+            end=None,
+            all_day=False,
+            time_notes="",
+            location=location,
+            links=self._parse_links(response.url, title),
+            source=self._parse_source(response),
+        )
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+        yield meeting
+
+    def _meeting_datetime(self, date, time):
+        meeting_start = date + " " + time
+        meeting_start = meeting_start.replace(", ", ",").strip()
+        return datetime.strptime(meeting_start, "%b %d,%Y %I:%M %p")
+
+    def _get_pdf_link(self, item):
+        pdf_tag = item.css("td:nth-child(4) > a")
+        if not (pdf_tag):
+            return None
+        pdf_link = pdf_tag[0].attrib["href"]
+        return pdf_link
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        try:
```

Comment:
Small thing, but if possible it would be good to replace "Comm" with "Committee" for committee meetings. To be safe we'd want to replace it only when it's at the end of the title, so something like this could work

```python
title = re.sub(r"Comm$", "Committee", title)
```